lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -80,6 +82,13 @@
     private BlockMetaData currentBlockMetadata;
     private long currentGroupRowCount;
     private long nextRowInGroup;
+    private int blockIndex = -1;
+    private BlockMetaData currentBlock;
+    /**
+     * Index of the first row of the current block within the Parquet data.
+     */
+    private long firstGlobalRowInGroup = -1;
```

Comment:
`firstRowIndexInGroup` and place it before `nextRowInGroup`

Reply:
I put in "global" and the comment to try to make it clear that it's counting a different index than `nextRowInGroup`. The comment alone is probably enough (though it's not as clear when the fields are used).

(Half of the refactors that used to be in this PR were attempts to make the field names clearer.)